### ISSUE TITLE
useSortedKeys has issues with nested objects

### ISSUE THREAD

**Issue by GiuseppeIII** [biomejs/biome#5848]
### Environment information

```bash
...
```

### Rule name

useSortedKeys

### Playground link

...

### Expected result

Been running into this a lot. The `useSortedKeys` action seems to have lots of issues with fixing the sorting on nested objects. As a basic example this:

```
const test = {
	aVar: 1,
	cVar: 2,
	bVar: {
		a: 3,
		b: 4,
		d: 6,
		c: 5,
	}
}
```

gets turned into 

```
const test = {
	aVar: 1,
	bVar: {
		a: 3,
		b: 4,
		d: 6,
		c: 5,
	}d6
	cVar: 2
}
```

I run into similar issues with almost all nested objects. 

Not sure how to trigger actions from the playground, but this should be pretty easily reproducible.

(Also as a side-note there is no Action bug report bug template)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I can't reproduce the issue

**Comment by GiuseppeIII** [comment]
I get this with simply
`biome.jsonc`
```
{
	"$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
	"assist": {
		"actions": {
			"source": {
				"useSortedKeys": "on"
			}
		}
	}
}
```

`test.ts`
```
const test = {
	aVar: 1,
	cVar: 2,
	bVar: {
		a: 3,
		b: 4,
		d: 6,
		c: 5,
	}
}
```

onSave

```
const test = {
	aVar: 1,
	bVar: {
		a: 3,
		b: 4,
		d: 6,
		c: 5,
	}d6
	cVar: 2
}
```

Don't think I can give a playground link as I don't think I can run actions from there. If really necessary I can create a repo, but it would literally just be the above two files.

**Comment by ematipico** [comment]
Sorry @GiuseppeIII but we need a reproduction. Use our interactive CLI. For example, at the beginning you omitted the fact that the error was caused when using the editor, which I didn't try.

**Comment by GiuseppeIII** [comment]
> For example, at the beginning you omitted the fact that the error was caused when using the editor, which I didn't try.

Ah yeah that is true, looks like the cli handles it perfectly. Wasn't even something I had thought to test 😅. I can spin up a quick repo.

**Comment by GiuseppeIII** [comment]
https://github.com/GiuseppeIII/biome-repro-1746214308439
